tmitchell/exercise.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/python3
+"""Run an interactive shell, users can view and edit a phone book and then save it to file."""
+class Person:
+    """Entry in phonebook able tyo store and display information"""
+    def __init__(self, name, hnumber, mnumber):
+        self.name = name
+        self.hnumber = hnumber
+        self.mnumber = mnumber
+
+    def display(self):
+        """Print stored information to screen."""
+        print("Name: %s" % (self.name))
+        print("Home Number: %s" % (self.hnumber))
+        print("Mobile Number: %s" % (self.mnumber))
+
+    def edit(self, new_name):
+        """Request user input and use it to edit entry details"""
+        if new_name != "":
+            self.name = new_name
+        else:
+            print("No changes will be made to the entry name")
+        print("Current entry home number:")
```

Comment:
A lot of repeated code below. You can abstract most of it into a function and just pass in `home` or `mobile` as parameter and return what's read.

Why is the input prompt printed separately and pass an empty string into `input`?